src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/HistoryCommand.kt

```diff
@@ -0,0 +1,24 @@
+package com.google.chip.chiptool.clusterclient.clusterinteraction
+
+import chip.clusterinfo.CommandResponseInfo
+import java.io.Serializable
+
+/**
+ * HistoryCommand stores information about each executed command and display
+ * necessary information for each item in [ClusterInteractionHistoryFragment]
+ */
+data class HistoryCommand(
+  val clusterName: String,
+  val commandName: String,
+  val parameterList: MutableList<HistoryParameterInfo>,
+  var responseValue: Map<CommandResponseInfo, Any>?,
+  var status: String?,
+  var endpointId: Int,
+  var deviceptr: Long,
```

Comment:
It probably makes more sense to store the device (node) ID, and then the fragment that consumes HistoryCommand calls `getConnectedDevicePointer(callback, nodeId)`. Storing the pointer is somewhat unstable.